### PR TITLE
add federated share type to sciencemeshShareProviderFactory

### PR DESCRIPTION
**Pull request by navid-dada**
I add a federated share provider  into sciencemeshShareProviderFactory to preserve functionality of federation sharing

### CODE REVIEW

**Comment by michielbdejong** [comment]
What is the rationale behind this? Do you want both regular internet-wide-OCM and OCM-within-ScienceMesh to work?
Or is it about aligning the share type we use for OCM-within-ScienceMesh with what code elsewhere expects?

**Comment by navid-dada** [comment]
It makes the ocm-internet sharing also available besides Reva, you can share your file with Reva or by federation approach. in the previous edition, you could not share files through the internet. but in the current edition, it became available again.

**Comment by mrvahedi68** [comment]
LGTM. But have some conflicts.

**Review by mrvahedi68 — APPROVED** [review]

**Comment by michielbdejong** [comment]
@navid-dada what is the status of this PR?

**Comment by navid-dada** [comment]
I just resolved the confilicts. but it seems that the changes will be modified by @parhamin2010 's changes.

**Comment by michielbdejong** [comment]
OK so what is your next step with this? Which issue does this fix? And is this still compatible with the thinking behind https://github.com/pondersource/sciencemesh-php/issues/126?

File: js/contacts.js [hunk 1]
```diff
@@ -1,41 +1,45 @@
...
-    } else {
+
+    var headerElement = document.getElementById("message");
     let token = JSON.parse(response);
+    if(response === '' || response === false || token['accepted_users'] === undefined) {
+        headerElement.innerHTML= 'No Reva Contact';
+        $('#test').show(); 
+    } else {
+        headerElement.innerHTML= 'Reva Contacts';
+
```
Comment (michielbdejong) [inline js/contacts.js:19]: These changes seem unrelated to "add federated share type to sciencemeshShareProviderFactory"?

File: js/contacts.js [hunk 2]
```diff
@@ -47,52 +51,15 @@ $.ajax({
                         </div>`;                  
                 var element = document.getElementById("test");
                 element.innerHTML = result;
-                $('#test_error').show();
-
+                $('#test').show();
```
Comment (michielbdejong) [inline js/contacts.js:54]: This looks like an unintentional revert and unrelated to "add federated share type to sciencemeshShareProviderFactory"?

File: js/contacts.js [hunk 3]
```diff
@@ -47,52 +51,15 @@ $.ajax({
...
-            } else {
-                console.log(result)
-            }
-        }
-     
-    }).fail(function (response, code) {
-        console.log(response)
-        //alert('The token is invalid')
-    });
-};
```
Comment (michielbdejong) [inline js/contacts.js:98]: Why remove this code block? Is it moved elsewhere?
These changes also seem unrelated to "add federated share type to sciencemeshShareProviderFactory"?

File: lib/AppInfo/Migrations/Version0001Date20211201101630.php [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+	public function __construct(IDBConnection $connection) {
+		$this->connection = $connection;
+	}
+
+	public function changeSchema(Schema $schema, array $options) {
+
+        $prefix = $options['tablePrefix'];
+
+        $shares = $schema->createTable("{$prefix}sciencemesh_shares");
```
Comment (michielbdejong) [inline lib/AppInfo/Migrations/Version0001Date20211201101630.php:25]: Oh wow, this introduces a new database table. Is this for incoming or for outgoing shares?
This is a big change, let's have a meeting about this first and take a decision about it first! I'll ping you and @parhamin2010 on Slack.

**Review by michielbdejong — CHANGES_REQUESTED** [review]

File: js/contacts.js (hunk 2 above)
Comment (navid-dada) [reply js/contacts.js:54]: I check the UI html code and other was not any tag with "test_error" id. so I concluded that I should remove first line.

File: lib/AppInfo/Migrations/Version0001Date20211201101630.php (hunk 4 above)
Comment (navid-dada) [reply lib/AppInfo/Migrations/Version0001Date20211201101630.php:25]: I cannot remember that I created this schema, I can remember there was an migration inside nc-sciencemesh and this schema is just a copy of that.

File: js/contacts.js (hunk 2 above)
Comment (michielbdejong) [reply js/contacts.js:54]: @parhamin2010 what is the correct element name now?

Comment (parhamin2010) [reply js/contacts.js:54]: It's id = test

**Comment by navid-dada** [comment]
we conclude that these changes are not necessary anymore.
